test/jdk/sun/security/pkcs11/tls/tls12/FipsModeTLS12.java

```diff
@@ -88,12 +89,11 @@ public static void main(String[] args) throws Exception {
         try {
             initialize();
         } catch (Exception e) {
-            System.out.println("Test skipped: failure during" +
-                    " initialization");
             if (enableDebug) {
                 System.out.println(e);
             }
-            return;
+            throw new SkippedException("Test skipped: failure during " +
+                    "initialization.");
         }
 
         if (shouldRun()) {
```

Comment:
you can remove the check at line 114-116 which checks `sunPKCS11NSSProvider`. It is set by call to `initialize` and fails if PKCS11 provider is not found. Let `shouldRun` fail if call to `getInstance` fails with any other reason than `NoSuchAlgorithmException`